vyxalbot2/__init__.py

```diff
@@ -482,66 +450,100 @@ async def runCommand(
-                    event.message_id,
-                    f"It was {random.choice(self.userDB.users())['name']}'s fault!",
-                )
+                return f"It was {random.choice(self.userDB.users())['name']}'s fault!"
             case "!good-bot":
-                await self.room.send(":3")
+                return await self.room.send(":3")
             case "hello":
-                await self.room.reply(
-                    event.message_id, random.choice(self.messages["hello"])
-                )
+                return random.choice(self.messages["hello"])
             case "goodbye":
-                await self.room.reply(
-                    event.message_id, random.choice(self.messages["goodbye"])
+                return random.choice(self.messages["goodbye"])
+
+    async def onEditMessage(self, room: Room, event: UnknownEvent):
+        myMessage: int = self.replyDB.getCorrespondingId(
+            event.args["message_id"]
+        )
+        try:
+            if match := re.fullmatch(
+                r"!!\/(?P<command>.+)", event.args["content"]
+            ):
+                rawCommand = match["command"]
+                for regex, command in COMMAND_REGEXES.items():
+                    if match := re.fullmatch(regex, rawCommand):
+                        response = self.runCommand(
+                            room, event, command, match.groupdict()
+                        )
+                        if response is not None:
+                            if isinstance(response, int):
+                                return response
+                            return await self.room.edit(
+                                myMessage,
+                                f":{myMessage}" + response,
+                            )
+                return await self.room.edit(
+                    myMessage,
+                    f"Sorry {event.user_name}, I'm afraid I can't do that.",
                 )
+        except Exception as e:
+            msg = f"@Ginger An error occurred while handling message {event.message_id}!"
+            await self.room.edit(myMessage, msg)
+            self.logger.exception(msg)
+            self.errorsSinceStartup += 1
 
     async def onMessage(self, room: Room, event: MessageEvent):
         try:
             if match := re.fullmatch(r"!!\/(?P<command>.+)", event.content):
                 rawCommand = match["command"]
                 for regex, command in COMMAND_REGEXES.items():
                     if match := re.fullmatch(regex, rawCommand):
-                        return await self.runCommand(
+                        response = self.runCommand(
                             room, event, command, match.groupdict()
                         )
+                        if response is not None:
+                            if isinstance(response, int):
+                                return response
+                            temp = await self.room.send(response)
```

Comment:
Imagine not using walrus here

Reply:
Imagine not being used to pre-3.8 python and never having really looked into walrus